doc/docs/Python_Tutorials/Adjoint_Solver.md

```diff
@@ -35,75 +34,86 @@ Python using [autograd](https://github.com/HIPS/autograd) as well as
 documentation](https://nlopt.readthedocs.io/en/latest/NLopt_Introduction/#equivalent-formulations-of-optimization-problems)
 for an overview of this approach. The minimax/epigraph approach is
 also covered in the [near-to-far field
 tutorial](https://nbviewer.org/github/NanoComp/meep/blob/master/python/examples/adjoint_optimization/06-Near2Far-Epigraph.ipynb).
 
-In this example, we use a minimum feature size of 150 nm for the line
-width *and* line spacing. The implementation of these constraints is
-based on [A.M. Hammond et al., Optics Express, Vol. 29, pp. 23916-38
+In this example, we use a minimum feature size of 150 nm for the
+linewidth *and* linespacing. The implementation of these constraints
+in the adjoint-solver module is based on [A.M. Hammond et al., Optics
+Express, Vol. 29, pp. 23916-38
 (2021)](https://doi.org/10.1364/OE.431188).
 
 There are five important items to note in the set up of the
 optimization problem:
 
-- The lengthscale constraint is activated only in the final epoch. It is
-   often helpful to mostly binarize the design before this final epoch. This is because
-  the lengthscale constraint forces binarization which could induce
-  large changes in an initial greyscale design and irrevocably spoil
-  the performance of the final design.
-
-- The initial value of the epigraph variable of the final epoch (in
-  which the linewidth constraint is imposed) should take into account
-  the value of the constraint itself to ensure a feasible starting
-  point.
-
-- The edge of the design region is padded by a filter radius (rather
+1. The lengthscale constraint is activated only in the final epoch. It
+   is often helpful to mostly binarize the design before this final
+   epoch. This is because the lengthscale constraint forces
+   binarization which could induce large changes in an initial
+   greyscale design and irrevocably spoil the performance of the final
+   design.
+
+2. The initial value of the epigraph variable of the final epoch (in
+  which the line width and spacing constraint is imposed) should take
+  into account the value of the constraint itself to ensure a feasible
+  starting point for the [method of moving asymptotes (MMA)
+  optimization
+  algorithm](https://nlopt.readthedocs.io/en/latest/NLopt_Algorithms/#mma-method-of-moving-asymptotes-and-ccsa),
+  which is based on the conservative convex separable approximation
+  (CCSA) algorithm.
+
+3. The edge of the design region is padded by a filter radius (rather
   than e.g., a single pixel) to produce measured lengthscales of the
   final design that are consistent with the imposed constraint.
 
-- The hyperparameters of the lengthscale constraint function (`glc` in
-  the script below), need to be chosen carefully to produce final
-  designs which do not significantly degrade the performance of the
-  unconstrained designs at the start of the final epoch.
+4. The hyperparameters of the lengthscale constraint function (`a1`,
```

Comment:
Eventually we should add context here... like what each of these actually does

Reply:
Acknowledged.